GraphClientFactory.md

```diff
@@ -18,7 +18,14 @@ There are two primary usage scenarios for the client factory.  Developers who wi
   - Set the SdkVersion header with the appropriate moniker based on the following structure `graph-{lang}-{version}`.  
 - Enable the developer to select a supported sovereign cloud using an enumerated list.  Selecting the sovereign cloud should ensure that the AuthenticationProvider uses the appropriate Authentication Endpoint.
 - Enable a developer to configure a HTTP proxy that will be used for outgoing requests.
-
+- Enable a developer provide custom URLs or endpoints. 
+  - `CustomURLs` option should be set on client creation.
+  - These URLs are different from the Graph service endpoints on the national clouds. 
+  - Certain workloads error out when an unexpected header is present in the request. The middlewares should check: 
+    - If the request URL is a Graph serve endpoint or a custom url provided by the developer, then append request headers or modify the request content.
+    - Else the middleware should delete request headers added by that middleware.
```

Comment:
Instead of delete request headers added by the middleware, can we just not add them?

Reply:
In case of redirection when we are using the same request context, the header remains and clearing it is safer.